services/workflows-service/src/workflow/workflow.service.ts

```diff
@@ -142,14 +142,41 @@ export class WorkflowService {
       extensions: true,
       persistStates: true,
       submitStates: true,
-      parentRuntimeDataId: true,
     };
 
     if (data.isPublic) {
-      return await this.workflowDefinitionRepository.createUnscoped({ data, select });
+      return await this.workflowDefinitionRepository.createUnscoped({
```

Comment:
Maybe `createFn = data.isPublic ? this.workflowDefinitionRepository.createUnscoped : this.workflowDefinitionRepository.create` instead this duplication?